mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -1067,6 +1072,13 @@ Status RuntimeClient::copyToHost(const MemRefValue &deviceMemRef,
 #endif
 }
 
+RuntimeClient::~RuntimeClient() {
+  // Reset the deleter for all tracked dlPackTensors
+  for (auto *tensor : dlPackTensors) {
+    tensor->deleter = nullptr;
```

Comment:
Deleting the deleter seems incorrect, since other memory not associated with the client gets freed there (eg strides and shape arrays). Instead, we should probably set the manager context handle to null here, then check for nullity in the deleter?